internal/nginx/nginx.go

```diff
@@ -399,3 +417,37 @@ func (nginx *Controller) UpdateIngressConfigFile(name string, cfg []byte) {
 	}
 	glog.V(3).Infof("The Ingress config file has been updated")
 }
+
+// UpdateConfigVersionFile writes the config version file.
+func (nginx *Controller) UpdateConfigVersionFile() {
+	cfg, err := nginx.verifyConfigGenerator.GenerateVersionConfig(nginx.configVersion)
+	if err != nil {
+		glog.Fatalf("Error generating config version content: %v", err)
+	}
+
+	filename := "/etc/nginx/config-version.conf"
+	tempname := "/etc/nginx/config-version.conf.tmp"
+	glog.V(3).Infof("Writing config version to %v", filename)
+
+	if bool(glog.V(3)) || nginx.local {
+		glog.Info(string(cfg))
+	}
+
+	if !nginx.local {
+		w, err := os.Create(tempname)
+		if err != nil {
+			glog.Fatalf("Failed to open %v: %v", filename, err)
+		}
+		_, err = w.Write(cfg)
+		if err != nil {
+			glog.Fatalf("Failed to write to %v: %v", filename, err)
+		}
+		w.Close()
```

Comment:
@isaachawley should you not `defer` right after `os.Create` above?

Reply:
Hmm. We don't normally do that. See [writing secrets here](https://github.com/nginxinc/kubernetes-ingress/blob/master/internal/nginx/nginx.go#L250). I'm open to changing it but I'd like to change them all at once.